sw/device/lib/dif/dif_otp_ctrl.h

```diff
@@ -0,0 +1,773 @@
+typedef struct dif_otp_ctrl { dif_otp_ctrl_params_t params; } dif_otp_ctrl_t;
+
+/**
+ * The result of a OTP operation.
+ */
+typedef enum dif_otp_ctrl_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifOtpCtrlOk = 0,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifOtpCtrlError = 1,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifOtpCtrlBadArg = 2,
+} dif_otp_ctrl_result_t;
+
+/**
+ * The result of a Direct Access Interface (DAI) operation.
+ */
+typedef enum dif_otp_ctrl_config_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifOtpCtrlConfigOk = kDifOtpCtrlOk,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifOtpCtrlConfigError = kDifOtpCtrlError,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifOtpCtrlConfigBadArg = kDifOtpCtrlBadArg,
+  /**
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  kDifOtpCtrlConfigLocked,
+} dif_otp_ctrl_config_result_t;
+
+/**
+ * The result of a Direct Access Interface (DAI) operation.
+ */
+typedef enum dif_otp_ctrl_dai_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifOtpCtrlDaiOk = kDifOtpCtrlOk,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifOtpCtrlDaiError = kDifOtpCtrlError,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifOtpCtrlDaiBadArg = kDifOtpCtrlBadArg,
+  /**
+   * Indicates that the DAI is busy and can't accept another incomming command.
+   */
+  kDifOtpCtrlDaiBusy,
+  /**
+   * Indicates that the given partition does not support this operation.
+   */
+  kDifOtpCtrlDaiBadPartition,
+  /**
+   * Indicates that the attempted DAI operation would go out of range of the
+   * requested partition.
+   */
+  kDifOtpCtrlDaiOutOfRange,
+  /**
+   * Indicates that the given address was not correctly aligned.
+   */
+  kDifOtpCtrlDaiUnaligned,
+} dif_otp_ctrl_dai_result_t;
+
+/**
+ * An OTP interrupt request type.
+ */
+typedef enum dif_otp_ctrl_irq {
+  /**
+   * Indicates that an asynchronous transaction completed.
+   */
+  kDifOtpCtrlIrqDone,
+  /**
+   * Indicates that an error has occured in the OTP controller.
+   */
+  kDifOtpCtrlIrqError,
+  // Your IRQs here!
+} dif_otp_ctrl_irq_t;
+
+/**
+ * A snapshot of the enablement state of the interrupts for OTP.
+ *
+ * This is an opaque type, to be used with the `dif_otp_ctrl_irq_disable_all()`
+ * and
+ * `dif_otp_ctrl_irq_restore_all()` functions.
+ */
+typedef uint32_t dif_otp_ctrl_irq_snapshot_t;
+
+/**
+ * A hardware-level status code.
+ */
+typedef enum dif_otp_ctrl_status {
```

Comment:
My plan was to not bother matching the hardware values and instead return a bitfield such that the caller would need to write something like
```c
status_bits & ~(1 << kMyStatus);
```
Is that too silly of an API, you think?

Reply:
You can still return a bitfield, and then have have the enum be individual bits in that bitfield, which would make the link between the bitfield you return, and the enum, clearer (because the bitfield you return would actually be of type `dif_otp_ctrl_status_t` - I think that would be ok semantics-wise)

Reply:
Assuming this question is resolved due to the note about the status struct.